integration/distributor_test.go

```diff
@@ -369,131 +1053,52 @@ overrides:
 
 				require.Equal(t, expResult, result)
 			}
+
+			for q, expected := range tc.metadataQueries {
+				result, err := client.GetPrometheusMetadata(q)
+				require.NoError(t, err)
+				defer result.Body.Close()
```

Comment:
Can we avoid `defer` in a loop? If done many enough times, it could lead to stack overflow. There must be a safer/cleaner solution?

Reply:
fixed